src/users/users.controller.ts

```diff
@@ -0,0 +1,58 @@
+import express from "express";
+import { ValidatedRequest } from "express-joi-validation";
+import {
+  UserAutoSuggestRequestSchema,
+  UserPatchRequestSchema,
+  UserPostRequestSchema,
+} from "../common/common.types";
+import { isEmpty } from "../common/common.utils";
+import usersService from "./users.service";
+
+class UsersController {
+  getUsers(
+    req: ValidatedRequest<UserAutoSuggestRequestSchema>,
+    res: express.Response
+  ) {
+    const login = isEmpty(req.query) ? "" : String(req.query.login);
+    const limit = isEmpty(req.query)
+      ? undefined
+      : typeof req.query.limit !== "number"
+      ? undefined
+      : req.query.limit;
+    const users = usersService.autosuggest(login, limit);
+    res.status(200).send(users);
```

Comment:
You are always using send() function. It is not a wrong usage, but to let you know send() does lots of things behind the scenes. I would suggest using the appropriate response method to be explicit. In this case .json().